lib/hawkular/base_client.rb

```diff
@@ -127,6 +127,28 @@ def generate_query_params(params = {})
       end
     end
 
+    # Generate a new url with the passed sufix path if the path is not already added
+    # also, this function always remove the slash at the end of the URL, so if your entrypoint is
+    # http://localhost/hawkular/inventory/ this function will return http://localhost/hawkular/inventory
+    # to the URL
+    # @param entrypoint [String] base path
+    # @param suffix_path [String] sufix path to be added if it doesn't exist
+    # @return [String] URL with path attached to it at the end
+    def normalize_entrypoint_url(entrypoint, suffix_path)
+      strip_path = suffix_path.gsub(%r{/$}, '')
+      strip_path.nil? || suffix_path = strip_path
+      strip_path = suffix_path.gsub(%r{^/}, '')
```

Comment:
I wonder it it would make sense to use the Uri class for this instead of doing all the (scaring :-) regex-magic.